### PR TITLE
Add dummy source file to SIMD to allow deduction of linker language.

### PR DESCRIPTION
**Pull request by crtrott**
This is really only necessary for Trilinos ...

This works with CMake prior to 3.19 which the other attempt didn't.

### CODE REVIEW

File: simd/src/KokkosSIMD_dummy.cpp [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// This file is need in order to get the linker language
```
Comment (masterleinad) [inline simd/src/KokkosSIMD_dummy.cpp:1]:
```suggestion
// This file is needed in order to get the linker language
```
Also, you need to fix the indentation.

**Comment by masterleinad** [comment]
You still need to fix the indentation.

**Comment by masterleinad** [comment]
```diff
+ ./scripts/docker/check_format_cpp.sh
diff --git a/simd/src/KokkosSIMD_dummy.cpp b/simd/src/KokkosSIMD_dummy.cpp
index 5d8fba3df..4f64ee0f6 100644
--- a/simd/src/KokkosSIMD_dummy.cpp
+++ b/simd/src/KokkosSIMD_dummy.cpp
@@ -2,6 +2,6 @@
 // for the header only submodule.
 // While we set the language properties in our normal cmake
 // path it does not get set in the Trilinos environment.
-// Furthermore, setting LINKER_LANGUAGE is only supported 
+// Furthermore, setting LINKER_LANGUAGE is only supported
 // in CMAKE 3.19 and up.
 void KOKKOS_SIMD_SRC_DUMMY_PREVENT_LINK_ERROR() {}
script returned exit code 1
```

**Comment by ndellingwood** [comment]
These changes resolved build errors with Trilinos (cuda/11.1 build, c++17 support) of he type:
```
nvlink fatal   : Could not open input file '../src/libkokkossimd.a'
make[2]: *** [kokkos/simd/unit_tests/KokkosSimd_UnitTest_SIMD.exe] Error 1
make[1]: *** [kokkos/simd/unit_tests/CMakeFiles/KokkosSimd_UnitTest_SIMD.dir/all] Error 2
```

**Review by dalg24 — APPROVED** [review]

File: simd/src/Kokkos_SIMD_dummy.cpp [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// This file is need in order to get the linker language
```
Comment (crtrott) [inline simd/src/Kokkos_SIMD_dummy.cpp:1]:
```suggestion
// This file is needed in order to get the linker language
```

**Review by masterleinad — APPROVED** [review]

**Pull request merged**